identity-wallet/src/verifiable_credential_record.rs

```diff
@@ -68,18 +74,40 @@ impl From<CredentialFormats<WithCredential>> for VerifiableCredentialRecord {
     }
 }
 
+fn get_achievement_name_from_data(credential_display: &serde_json::Value) -> Option<String> {
+    let cred_subject = credential_display.get("credentialSubject")?;
+    let achievement = cred_subject.get("achievement")?;
+    let name = achievement.get("name")?;
+
+    // Don't use the to_string for the name variable as it add's "" around the string
+    name.as_str().map(|name| name.to_string())
+}
+
+fn get_type_name_from_data(credential_display: &serde_json::Value) -> Option<String> {
+    match credential_display.get("type")? {
+        // Don't use the to_string for the name variable as it add's "" around the string
+        Value::Array(array) => array.last()?.as_str().map(|name| name.to_string()),
+        _ => None,
+    }
+}
+
 #[derive(Debug, Serialize, Deserialize, Clone, Eq, Derivative, TS)]
 #[derivative(PartialEq)]
 #[ts(export, export_to = "bindings/display-credential/DisplayCredential.ts")]
 pub struct DisplayCredential {
+    /// Also display_icon
```

Comment:
I don't think we need to comment that the `id` is also used somewhere else as it's not relevant for the `DisplayCredential`. Also, it's not "display_icon", rather "image id".

Reply:
Well it just makes everything easier. Because last time we took a long time finding out where to find the "image id" and we had to look through the code until you realized it was that field.

Reply:
Let's make a proper comment out of it then such as "This id is also used as the image asset id"